### PR TITLE
Add acceptance tests using ANYCAST routing method

### PR DESCRIPTION
**Pull request by alarbada**
### Description

Adds an implementation of a conduit connector that abstracts activemq Artemis using the STOMP protocol.
The acceptance tests use the ANYCAST routing method.
Here's some info about the different routing methods:
https://activemq.apache.org/components/artemis/documentation/latest/address-model.html#basic-manual-configuration

### Missing

- [ ] Docs
- [ ] TLS configuration + tests (the code is there but I haven't tested it locally)
- [ ] acceptance tests testing the MULTICAST routing method.
- [x] Improve github actions

### CODE REVIEW

File: config.go
Comment (hariso) [inline config.go:1]: This appears similar to the `shared.go` file in the AMQ Classic connector. I believe we should apply the same changes here too.

File: acceptance_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+			DestinationConfig: destinationConfig,
+			BeforeTest: func(t *testing.T) {
+				destination := uniqueDestinationName(t)
+				sourceConfig["destination"] = destination
+				destinationConfig["destination"] = destination
+				destinationConfig["destinationHeader"] = destination
+			},
+			Skip: []string{
+				"TestSource_Configure_RequiredParams",
+				"TestDestination_Configure_RequiredParams",
```
Comment (hariso) [inline acceptance_test.go:54]: If we really need to skip some tests, we should mentioned the reason for that.

File: acceptance_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			SourceConfig:      sourceConfig,
+			DestinationConfig: destinationConfig,
+			BeforeTest: func(t *testing.T) {
+				destination := uniqueDestinationName(t)
+				sourceConfig["destination"] = destination
+				destinationConfig["destination"] = destination
+				destinationConfig["destinationHeader"] = destination
+			},
+			Skip: []string{
+				// Configure tests are faulty since we rely on paramgen to validate required parameters.
```
Comment (hariso) [inline acceptance_test.go:56]: Hmm, how do you mean faulty? An issue with the connector SDK?

File: source.go [hunk 3]
```diff
@@ -1,98 +1,231 @@
...
+		return fmt.Errorf("message with ID %q not found", pos.MessageID)
+	}
+
+	if err := s.conn.Ack(msg); err != nil {
+		return fmt.Errorf("failed to ack message: %w", err)
+	}
+
+	_, exists := s.storedMessages.Pop(pos.MessageID)
+	if !exists {
+		sdk.Logger(ctx).Trace().Str("messageID", pos.MessageID).Msg("message was already acked")
```
Comment (hariso) [inline source.go:195]: Can it ever come to this actually?

File: acceptance_test.go (hunk 2 above)
Comment (alarbada) [reply acceptance_test.go:56]: Basically the root of the issue ignoring these two is that both `source.Configure` and `destination.Configure` don't validate the given configuration. The idea that I got was that validation with struct tags like `required` or `inclusion=val1|val2`  was done in an upper layer, not by the Configure method.

I'd be happy to validate them again, would prevent me silly mistakes developing acceptance tests, but I don't see any function on the sdk that used the `map[string]sdk.Parameter` to validate.

File: source.go (hunk 3 above)
Comment (alarbada) [reply source.go:195]: woops yeah, silly mistake :sweat_smile:

**Pull request merged**